Completion of JavassistParameterDeclaration

kgeilmann opened the pull request:
This PR completes the implementation of JavassistParameterDeclaration. 
For the isXYZ methods it copies the behaviour of ReflectionParameterDeclaration and if available the parameter name gets read from the jar.

kgeilmann committed:
Adding some of the missing implementations of JavassistParameterDeclaration, mainly getName

kgeilmann committed:
Reverting accidently commited change in pom.xml

ftomassetti commented on javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistUtils.java:
```diff
@@ -206,4 +203,19 @@ private static ResolvedType typeArgumentToType(SignatureAttribute.TypeArgument t
+                return attr.variableName(paramNumber + pos);
+            }
+        }
+        return null;
```
We should explain while this return null

matozoid replied:
Nothing in JavaParser should return null really. Use Optional.

ftomassetti commented on javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistParameterDeclarationTest.java:
```diff
@@ -0,0 +1,168 @@
+    }
+
+    @Test
+    public void noNamesAvailable_inInterfaceMethods() {
```
I think we should not use underscores in names

ftomassetti reviewed (approved):
To me it seems good and a very valuable contribution: thanks. I have just minor comments

matozoid committed:
Merge branch 'master' into javassistparamdecl

kgeilmann committed:
Adding javadoc comment to explain why the name might not be available.

Changed return value from String to Optional to avoid null. Renamed tests to conform to code style.

matozoid committed:
Merge branch 'master' into javassistparamdecl

matozoid committed:
Merge branch 'master' into javassistparamdecl
